common/data/src/main/java/org/thingsboard/server/common/data/notification/info/AlarmNotificationInfo.java

```diff
@@ -41,25 +40,28 @@ public class AlarmNotificationInfo implements RuleOriginatedNotificationInfo {
     private UUID alarmId;
     private EntityId alarmOriginator;
     private String alarmOriginatorName;
+    private String alarmOriginatorLabel;
     private AlarmSeverity alarmSeverity;
     private AlarmStatus alarmStatus;
     private boolean acknowledged;
     private boolean cleared;
     private CustomerId alarmCustomerId;
     private DashboardId dashboardId;
+    private Map<String, String> details;
 
     @Override
     public Map<String, String> getTemplateData() {
-        return mapOf(
-                "alarmType", alarmType,
-                "action", action,
-                "alarmId", alarmId.toString(),
-                "alarmSeverity", alarmSeverity.name().toLowerCase(),
-                "alarmStatus", alarmStatus.toString(),
-                "alarmOriginatorEntityType", alarmOriginator.getEntityType().getNormalName(),
-                "alarmOriginatorName", alarmOriginatorName,
-                "alarmOriginatorId", alarmOriginator.getId().toString()
-        );
+        Map<String, String> templateData = new HashMap<>(details);
```

Comment:
This is not safe. Existing notifications (possibly scheduled) don't have the `details`. Account for null